lib/inc/drogon/plugins/Redirector.h

```diff
@@ -0,0 +1,120 @@
+/**
+ *  @file Redirector.h
+ *  @author An Tao
+ *
+ *  Copyright 2018, An Tao.  All rights reserved.
+ *  https://github.com/an-tao/drogon
+ *  Use of this source code is governed by a MIT license
+ *  that can be found in the License file.
+ *
+ *  Drogon
+ *
+ */
+
+#pragma once
+
+#include <drogon/plugins/Plugin.h>
+#include <trantor/net/InetAddress.h>
+#include <drogon/HttpRequest.h>
+#include <vector>
+
+namespace drogon
+{
+namespace plugin
+{
+/**
+ * @brief The RedirectorHandler is a function object that can be registered to
+ * the Redirector plugin. It is used to redirect requests to proper URLs. Users
+ * can modify the protocol, host and path of the request. If a false value is
+ * returned, the request will be considered as invalid and a 404 response will
+ * be sent to the client.
+ */
+using RedirectorHandler =
+    std::function<bool(const drogon::HttpRequestPtr &,
+                       std::string &,  //"http://" or "https://"
+                       std::string &,  // host
+                       bool &)>;       // path changed or not
```

Comment:
The pathChanged parameter will not be used, we can use pathRewriter handler to make changes to the path.

Reply:
Maybe some redirectors need to change all three things at the same time.